pkg/controllers/dnspolicy/dnspolicy_healthchecks.go

```diff
@@ -32,21 +33,51 @@ func (r *DNSPolicyReconciler) reconcileHealthCheckProbes(ctx context.Context, dn
 		}
 	}
 
+	gwsWithIgnoredHosts := []string{}
 	// Reconcile DNSHealthCheckProbes for each gateway directly referred by the policy (existing and new)
 	for _, gw := range append(gwDiffObj.GatewaysWithValidPolicyRef, gwDiffObj.GatewaysMissingPolicyRef...) {
 		log.V(3).Info("reconciling probes", "gateway", gw.Name)
-		expectedProbes := r.expectedHealthCheckProbesForGateway(ctx, gw, dnsPolicy)
+		expectedProbes, ignoredHosts := r.expectedHealthCheckProbesForGateway(ctx, gw, dnsPolicy)
 		if err := r.createOrUpdateHealthCheckProbes(ctx, expectedProbes); err != nil {
 			return fmt.Errorf("error creating or updating expected probes for gateway %v: %w", gw.Gateway.Name, err)
 		}
 		if err := r.deleteUnexpectedGatewayHealthCheckProbes(ctx, expectedProbes, gw.Gateway, dnsPolicy); err != nil {
 			return fmt.Errorf("error removing unexpected probes for gateway %v: %w", gw.Gateway.Name, err)
 		}
-
+		if ignoredHosts != nil {
+			gwsWithIgnoredHosts = append(gwsWithIgnoredHosts, gw.Namespace+"/"+gw.Name)
+		}
 	}
+
+	addHealthCheckCondition(dnsPolicy, gwsWithIgnoredHosts)
```

Comment:
so I can see this is iterating through gateways, but i think this is to allow for a policy changing targetRef. A policy can only ever target one gateway. So do we want to add a condition if the ref is no longer valid?